### DISCUSSION TITLE
error: 'tft' was not declared in this scope

### DISCUSSION THREAD

**Discussion by 2ant** [Bodmer/TFT_eSPI#1795] [Q&A - General]
Hello,

I'm having trouble tring to order my code from spaghetti to nice .cpp and .h files.

I thought that puting : 

```ruby

#include <TFT_eSPI.h>

TFT_eSPI tft = TFT_eSPI();
```

At the start before the setup made so tft was global and accessible from anywhere.

I have a Functions.cpp file with : 

```ruby
#include <TFT_eSPI.h>
#include "Functions.h"

void ActionMessage::UpdateMessage(int BoxNum){
  tft.fillRect(100, 75, 100, 20, TFT_CYAN);
  tft.setCursor(100,75);
  tft.print(BoxNum);
};
```

I got this error : \Functions.cpp:8:3: error: 'tft' was not declared in this scope

I don't know what to do.

**Comment by Pablo2048** [comment]
You have missing public declaration of `tft` - just put in Functions.cpp after all include this
```cpp
extern TFT_eSPI tft;
```

**Reply by iCodeAlot1001** [reply]
thanks my code now compiled
